### PR TITLE
[Merged by Bors] - feat: add compactness of the spectrum as an instance

### PR DESCRIPTION
**Pull request by j-loreaux**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Algebra/Spectrum.lean [hunk 1]
```diff
@@ -141,6 +141,18 @@ theorem mem_resolventSet_iff {r : R} {a : A} : r ∈ resolventSet R a ↔ IsUnit
   Iff.rfl
 #align spectrum.mem_resolvent_set_iff spectrum.mem_resolventSet_iff
 
+theorem algebraMap_mem_iff (S : Type*) {R A : Type*} [CommSemiring R] [CommSemiring S]
```
Comment (urkud) [inline Mathlib/Algebra/Algebra/Spectrum.lean:144]: Should these 2 lemmas be `@[simp]`?

Comment (urkud) [reply Mathlib/Algebra/Algebra/Spectrum.lean:144]: Otherwise LGTM.

Comment (j-loreaux) [reply Mathlib/Algebra/Algebra/Spectrum.lean:145]: @urkud I'm not sure, but we'll say yes for now.

**Comment by urkud** [comment]
Thanks! :tada:
bors merge

**Pull request closed without merging**